src/condition.cpp

```diff
@@ -1887,65 +1888,62 @@ void ConditionLight::serialize(PropWriteStream& propWriteStream)
 	propWriteStream.write<uint32_t>(lightChangeInterval);
 }
 
-void ConditionSpellCooldown::addCondition(Creature* creature, const Condition* condition)
+void ConditionSpellCooldown::addCondition(std::shared_ptr<Creature> creature, const Condition* condition)
 {
 	if (updateCondition(condition)) {
 		setTicks(condition->getTicks());
 
 		if (subId != 0 && ticks > 0) {
-			Player* player = creature->getPlayer();
-			if (player) {
+			if (auto player = creature->getPlayer()) {
```

Comment:
Could this be `const auto`
```suggestion
			if (const auto player = creature->getPlayer()) {
```